src/components/Button/Button.tsx

```diff
@@ -202,34 +202,37 @@ export function Button(props: BaseProps & ButtonElProps): ReturnType;
 export function Button({
   children,
   icon: Icon,
+  enableTracking = true,
   ...props
 }: ButtonProps): ReturnType {
   const { Link } = useComponents();
   const LinkButton = BaseButton.withComponent(Link);
   const ButtonElement = props.href ? LinkButton : BaseButton;
 
-  /** Set default value for tracking dispatch */
-  const { label = undefined, component = 'button', data = undefined } =
-    props.trackingPayload || {};
-  const hasTracking = !!label || props.enableTracking;
+  // Set default value for tracking dispatch
+  const { label, component = 'button', customParameters } =
+    props.tracking || {};
+  // When there's no label provided, we can still dispath
+  // tracking event by set enableTracking to true.
```

Comment:
Is the `label` a required property for tracking events?

Reply:
No. We mark it as optional in collector interface.

Reply:
Sorry. My bad! Please refers to https://sumup.slack.com/archives/CM9JBB5GE/p1591184697010000
It should be a required property for tracking event.

Reply:
In that case, we could get rid of the `enableTracking` prop and simply dispatch the event when any tracking data is provided in the `tracking` prop, right?

Reply:
The downside is that there's no automatic tracking out of the box, but it forces developers to add tracking data which should improve the quality of the overall data.

Reply:
@connor-baer We are avoiding automatic tracking since there is no automated way to ensure an id for the buttons for instance. I'm happy to expand in this topic if required